dao-voting/tests/harness.rs

```diff
@@ -0,0 +1,897 @@
+                .deposit()
+                .tx_params(tx_params)
+                .call_params(call_params)
+                .call()
+                .await
+                .unwrap()
+                .value;
+        }
+
+        #[tokio::test]
+        #[should_panic]
+        async fn panics_when_not_initialized() {
+            let (_gov_token, gov_token_id, deployer, user, asset_amount) = setup().await;
+
+            assert!(
+                deployer
+                    .gov_token
+                    .unwrap()
+                    .mint_and_send_to_address(100, user.wallet.address())
+                    .append_variable_outputs(1)
+                    .call()
+                    .await
+                    .unwrap()
+                    .value
+            );
+
+            let tx_params = TxParameters::new(None, Some(1_000_000), None, None);
+            let call_params = CallParameters::new(
+                Some(asset_amount),
+                Some(AssetId::from(*gov_token_id)),
+                Some(100_000),
+            );
+            user.dao_voting
+                .deposit()
+                .tx_params(tx_params)
+                .call_params(call_params)
+                .call()
+                .await
+                .unwrap()
+                .value;
+        }
+
+        #[tokio::test]
+        #[should_panic]
+        async fn panics_with_incorrect_asset() {
+            let (_gov_token, gov_token_id, deployer, user, asset_amount) = setup().await;
+
+            let another_asset_id = Contract::deploy_with_salt(
+                "./tests/artifacts/asset/out/debug/asset.bin",
+                &deployer.wallet,
+                TxParameters::default(),
+                Salt::from([1u8; 32]),
+            )
+            .await
+            .unwrap();
+
+            let another_asset =
+                GovToken::new(another_asset_id.to_string(), deployer.wallet.clone());
+
+            assert!(
+                another_asset
+                    .mint_and_send_to_address(100, user.wallet.address())
+                    .append_variable_outputs(1)
+                    .call()
+                    .await
+                    .unwrap()
+                    .value
+            );
+
+            deployer
+                .dao_voting
+                .constructor(gov_token_id)
+                .call()
+                .await
+                .unwrap()
+                .value;
+
+            let tx_params = TxParameters::new(None, Some(1_000_000), None, None);
+            let call_params = CallParameters::new(
+                Some(asset_amount),
+                Some(AssetId::from(*gov_token_id)),
+                Some(100_000),
+            );
+            user.dao_voting
+                .deposit()
+                .tx_params(tx_params)
+                .call_params(call_params)
+                .call()
+                .await
+                .unwrap()
+                .value;
+        }
+    }
+}
+
+mod vote {
```

Comment:
Can you order the modules to be in the same order as the abi functions? It makes it easier to follow when reviewing. It can easily take 2h to review something just a single time let alone multiple times. It makes it difficult to catch everything and even more so if things are out of order and I have to keep jumping around different files and within the same file

Reply:
I think it took me 3h+ to go through it this time mainly because of the tests...